libs/design/src/molecules/list/list/list.component.ts

```diff
@@ -32,4 +51,36 @@ export class DaffListComponent {
   @HostBinding('class.daff-list--navigation') get navigation() {
     return this.mode === DaffListModeEnum.Navigation;
   }
+  
+  listType: DaffListType;
+
+  constructor(private elementRef: ElementRef, private renderer: Renderer2) {}
+
+  ngOnInit() {
+    for (const attr of LIST_HOST_ATTRIBUTES) {
+      if (this._hasHostAttributes(attr)) {
+        this.listType = attr;
+      }
+    }
+  }
+
+  @HostBinding('class.daff-nav-list') get nav() {
+    return this.listType === DaffListTypeEnum.Nav;
+  }
+
+  /**
+   * Sets the role for a `<daff-nav-list>` to navigation.
+   */
+  @HostBinding('attr.role') get role() {
+    return this.listType === DaffListTypeEnum.Nav ? 'navigation' : 'list';
+  };
+
+  _getHostElement() {
+    return this.elementRef.nativeElement;
+  }
+
+  /** Gets whether a list has one of the given attributes. */
+  _hasHostAttributes(...attributes: string[]) {
```

Comment:
`_hasHostAttributes` is a bad name for what this function does. This checks the name of the host element itself (`daff-list` vs. `daff-nav-list`).

Reply:
Remove this from simplifying the code